ocaml/libs/http-lib/http_svr.ml

```diff
@@ -359,6 +359,12 @@ let request_of_bio_exn ~proxy_seen ~read_timeout ~total_timeout ~max_length bio
     proxy |> Option.fold ~none:[] ~some:(fun p -> [("STUNNEL_PROXY", p)])
   in
   let open Http.Request in
+  (* Below transformation only keeps one value per key, whereas
+     a fully compliant implementation following Uri's interface
+     would operate on list of values for each key instead *)
+  let kvlist_flatten ls =
+    List.map (function k, v :: _ -> (k, v) | k, [] -> (k, "")) ls
```

Comment:
One thing to keep in mind is that `Uri.of_string` seems to be more accepting in terms of the query string than `parse_keyvalpairs`, as it even accepts empty values. These would have resulted in an exception, but here we turn them into `""` values. We could raise `Http_parse_failure` in this case, as the old function would.

Reply:
I think `parse_keyvalpairs` also accepts empty values:
```ocaml
utop # parse_keyvalpairs "a=c&b=&e=";;
- : (string * string) list = [("a", "c"); ("b", ""); ("e", "")]
```
Is this what you mean?

As a matter of fact, I'm not quite sure how to trigger the `| []` match hand in `parse_keyvalpairs` at all:
```ocaml
utop # parse_keyvalpairs "";;
- : (string * string) list = [("", "")]
```

Reply:
If `Astring.string.cuts`'s optional parameter is changed to `~empty:false`, then:

```ocaml
utop # parse_keyvalpairs "&";;
- : (string * string) list = []
utop # parse_keyvalpairs "=";;
HTTP_PARSE_FAILURE
utop # parse_keyvalpairs "k=";;
- : (string * string) list = [("k", "")]
```

Reply:
You're right! That means `kvlist_flatten` is good as it is.

And this is a bit counter-intuitive:
```
utop # Astring.String.cuts ~sep:"=" "";;
- : string list = [""]
```